packages/core/src/wallet/IndyWallet.ts

```diff
@@ -279,95 +264,85 @@ export class IndyWallet implements Wallet {
+      )
+      const unpackedMessage = JsonEncoder.fromBuffer(unpackedMessageBuffer)
+      return {
+        senderVerkey: unpackedMessage.sender_verkey,
+        recipientVerkey: unpackedMessage.recipient_verkey,
+        message: JsonEncoder.fromString(unpackedMessage.message),
+      }
+    } catch (error) {
+      throw new WalletError('Error unpacking message', { cause: error })
     }
   }
 
-  public async sign(data: Buffer, verkey: Verkey): Promise<Buffer> {
-    const signatureBuffer = await this.indy.cryptoSign(this.walletHandle, verkey, data)
-
-    return signatureBuffer
-  }
-
-  public async verify(signerVerkey: Verkey, data: Buffer, signature: Buffer): Promise<boolean> {
-    // check signature
-    const isValid = await this.indy.cryptoVerify(signerVerkey, data, signature)
-
-    return isValid
-  }
-
-  public async addWalletRecord(type: string, id: string, value: string, tags: Record<string, string>) {
-    return this.indy.addWalletRecord(this.walletHandle, type, id, value, tags)
-  }
-
-  public async updateWalletRecordValue(type: string, id: string, value: string) {
-    return this.indy.updateWalletRecordValue(this.walletHandle, type, id, value)
-  }
-
-  public async updateWalletRecordTags(type: string, id: string, tags: Record<string, string>) {
-    return this.indy.addWalletRecordTags(this.walletHandle, type, id, tags)
-  }
-
-  public async deleteWalletRecord(type: string, id: string) {
-    return this.indy.deleteWalletRecord(this.walletHandle, type, id)
-  }
+  public async sign(data: Buffer, verkey: string): Promise<Buffer> {
+    try {
+      const signatureBuffer = await this.indy.cryptoSign(this.walletHandle, verkey, data)
 
-  public async search(type: string, query: WalletQuery, options: WalletSearchOptions) {
-    const sh: number = await this.indy.openWalletSearch(this.walletHandle, type, query, options)
-    const generator = async function* (indy: typeof Indy, wh: number) {
-      try {
-        while (true) {
-          // count should probably be exported as a config?
-          const recordSearch = await indy.fetchWalletSearchNextRecords(wh, sh, 10)
-          for (const record of recordSearch.records) {
-            yield record
-          }
-        }
-      } catch (error) {
-        // pass
-      } finally {
-        await indy.closeWalletSearch(sh)
-      }
+      return signatureBuffer
+    } catch (error) {
+      throw new WalletError(`Error signing data with verkey ${verkey}`, { cause: error })
     }
-
-    return generator(this.indy, this.walletHandle)
   }
 
-  public getWalletRecord(type: string, id: string, options: WalletRecordOptions): Promise<WalletRecord> {
-    return this.indy.getWalletRecord(this.walletHandle, type, id, options)
-  }
+  public async verify(signerVerkey: string, data: Buffer, signature: Buffer): Promise<boolean> {
+    try {
+      // check signature
+      const isValid = await this.indy.cryptoVerify(signerVerkey, data, signature)
```

Comment:
Same here.

Reply:
I think in this case `isValid` adds some useful context to what `cryptoVerify` (and `verify()`) returns